app/api/deps.py

```diff
@@ -28,30 +29,36 @@ def get_current_user(
         db: Session = Depends(get_db),
         token: str = Depends(reusable_oauth2)
 ) -> User:
-    try:
-        payload = jwt.decode(
-            token, settings.JWT_KEY, algorithms=[security.ALGORITHM]
-        )
-    except (jwt.PyJWTError, ValidationError):
-        raise HTTPException(
-            status_code=status.HTTP_403_FORBIDDEN,
-            detail="Could not validate credentials",
+
+    if settings.USING_GATEKEEPER:
+        response = requests.post(
+            url=settings.GATEKEEPER_BASE_URL.unicode_string() + "api/validate_token/",
+            headers={"Content-Type": "application/json"},
+            json={"token": token, "token_type": "access"}
         )
-    user_db = user.get(db, id=payload["sub"])
-    if not user_db:
-        raise HTTPException(status_code=404, detail="User not found")
-    return user_db
 
+        if response.status_code == 400:
+            raise HTTPException(
+                status_code=400,
+                detail="Error, bad jwt token."
+            )
 
-def get_administrator(
-        db: Session = Depends(get_db),
-        token: str = Depends(reusable_oauth2)
-) -> User:
-    user_db = get_current_user(db=db, token=token)
+        return User(email=token)
```

Comment:
email=token?

Reply:
Yeah, since the get_current_user func returns a User obj. I store the token, in case of gatekeeper, as the username.

This isn't the best solution, but it works for now since that function is used to check for a logged in user, and the fields aren't used anywhere.

This will definitely need to be revised later once/if we add rbac or abac.

Reply:
@vlf-stefan-drobic , @fedjo  user instance is not used anywhere, we use it only for dependency call to make API locked, we care here about token (gatekeeper cares about user info). I believe we do not need to return user instance